src/types/Time.ts

```diff
@@ -0,0 +1,33 @@
+import * as TypeUtils from '../TypeUtils'
+
+export const Time: SBVRType<string | null, NullableDate> = {
+	types: {
+		postgres: 'TIME',
+		mysql: 'TIME',
+		websql: 'TEXT',
+		odata: {
+			name: 'Edm.DateTime'
+		},
+	},
+
+	fetchProcessing: (data, callback) => {
+		let date: NullableDate
+		if (data !== null) {
+			// We append the date of the epoch so that we can parse this as a valid date.
+			date = new Date('Thu, 01 Jan 1970 ' + data)
+		} else {
+			date = null
+		}
+		callback(null, date)
+	},
+
+	validate: (value, required, callback) => {
+		TypeUtils.validate.date(value, required, (err, response) => {
+			if (err) {
+				callback(err)
+			} else {
+				callback(null, response!.toLocaleTimeString())
```

Comment:
I'm pretty sure that `response` can be null here, eg if the time field in the db is nullable and a null is stored there - so this needs to check for that (assuming I'm right).  That said it's not a new bug, but it is one that typescript highlights.  Adding a test for this would also be good

Reply:
@Page- that should be check by looking at `required` right? Why don't we do that for other validation checks too? Some of them will silently convert to a default value in the validation function if the supplied value is null